plugins/org.eclipse.elk.alg.vertiflex/src/org/eclipse/elk/alg/vertiflex/p2relative/InverseYNodeComparator.java

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Kiel University and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 
+ *******************************************************************************/
+package org.eclipse.elk.alg.vertiflex.p2relative;
+
+import org.eclipse.elk.alg.vertiflex.InternalProperties;
+import org.eclipse.elk.graph.ElkNode;
+
+/**
+ * Comparator for sorting ElkNodes according to their y positions in descending order and model order.
+ *
+ */
+class InverseYNodeComparator implements java.util.Comparator<ElkNode> {
+
+    @Override
+    public int compare(final ElkNode a, final ElkNode b) {
+
+        int sortYresult =  Double.compare(b.getY(), a.getY());
+        if (sortYresult == 0) {
+            int intSortresult =  Integer.compare(a.getProperty(InternalProperties.NODE_MODEL_ORDER), 
```

Comment:
Would it make sense to move the MODEL_ORDER property into a more general place to be used here?